exporter/prometheusexporter/accumulator.go

```diff
@@ -204,6 +209,12 @@ func (a *lastValueAccumulator) accumulateSum(metric pmetric.Metric, il pcommon.I
 			continue
 		}
 
+		// Delta-to-Cumulative
+		if doubleSum.AggregationTemporality() == pmetric.MetricAggregationTemporalityDelta {
+			ip.SetStartTimestamp(mv.value.Sum().DataPoints().At(0).StartTimestamp())
+			ip.SetIntVal(ip.IntVal() + mv.value.Sum().DataPoints().At(0).IntVal())
```

Comment:
Does this line need to support float64-valued counters separately?

Reply:
Yeah  the Prometheus counter is float64-based, so I think we should do that. 

Should it be:

```
ip.SetIntVal(ip.IntVal() + mv.value.Sum().DataPoints().At(0).IntVal())
ip.SetDoubleVal(ip.DoubleVal() + mv.value.Sum().DataPoints().At(0).DoubleVal())
```

Or we must parse the IntVal to the DoubleVal type then add them up?

Or just do a simple if/else to check the type of the current delta value/last cumulative value was Int/Double?

Regarding to this, let me find if any other receiver produced the delta counter as StatsD receiver will always parse it's counters to Int.

Reply:
I see -- what you have is correct for statsd to PRW, but I was thinking of other OTLP senders (e.g., an SDK) configured for delta temporality that might use floating point. There is a `pdata.MetricValueType` that indicates what the incoming point has. I would say that since PRW exports floating points always, possibly the best solution is to convert points from integer (if present) to double somewhere above the accumulator, so that the stored point is always a floating point.

Reply:
Roger. Working on this :loading: